### PR TITLE
Add `shadows_disabled` macro in Compatibility renderer

### LINKED ISSUE #84386: Shadows cannot be disabled in the material in the Compatibility renderer
### Godot version

Godot v4.2.beta (b733901e9)

### System information

Godot v4.2.beta (b733901e9) - Solus 4.4 - X11 - GLES3 (Compatibility) - Mesa Intel(R) HD Graphics 3000 (SNB GT2) () - Intel(R) Core(TM) i5-2500K CPU @ 3.30GHz (4 Threads)

### Issue description

The `shadows_disabled` render mode doesn't work at all with the Compatibility renderer, it simply isn't implemented.

### Steps to reproduce

-add any two MeshInstances to a 3D scene
-add a light so that shadow falls on one of the meshes
-add a new shader material to the shadowed mesh and create a shader
-add `render_mode shadows_disabled;` to the shader
-shadows are still visible

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by jsjtxietian**
Fixes #84386

### CODE REVIEW

**Review by clayjohn — APPROVED** [review]
Seems fine to me. 

Let's merge this for 4.3 as we are in the final stages of 4.2. We can cherrypick this for 4.2.1

File: drivers/gles3/shaders/scene.glsl [hunk 1]
```diff
@@ -593,11 +593,11 @@ layout(std140) uniform GlobalShaderUniformData { //ubo:1
 	vec4 global_shader_uniforms[MAX_GLOBAL_SHADER_UNIFORMS];
 };
 
-	/* Material Uniforms */
++ /* Material Uniforms */
```
Comment (AThousandShips) [inline drivers/gles3/shaders/scene.glsl:596]:
```suggestion
/* Material Uniforms */
```

File: drivers/gles3/shaders/scene.glsl [hunk 2]
```diff
@@ -593,11 +593,11 @@ layout(std140) uniform GlobalShaderUniformData { //ubo:1
...
 
 #ifdef MATERIAL_UNIFORMS_USED
 
-/* clang-format off */
+		/* clang-format off */
```
Comment (AThousandShips) [inline drivers/gles3/shaders/scene.glsl:600]:
```suggestion
/* clang-format off */
```

**Review by AThousandShips — COMMENTED** [review]
Some accidental changes that should be restored

File: drivers/gles3/shaders/scene.glsl (hunk 1 above)
Comment (jsjtxietian) [reply drivers/gles3/shaders/scene.glsl:596]: Sorry, I happen to upgrade my local llvm version so clang-format version is too high.

**Review by Calinou — CHANGES_REQUESTED** [review]
Tested locally. This PR breaks rendering in an empty project with a few meshes, even if they have no material assigned: [test_pr_84416.zip](https://github.com/godotengine/godot/files/13279426/test_pr_84416.zip)

```
ERROR: SceneShaderGLES3: Fragment shader compilation failed:
0(1540) : error C7102: unmatched #endif

   at: _display_error_with_code (drivers/gles3/shader_gles3.cpp:252)
```

**Comment by jsjtxietian** [comment]
Hi @Calinou Did you use the last commit, I do miss a macro there when try to fix the clang format issue.

**Review by Calinou — APPROVED** [review]
Tested latest revision, it works as expected. Enabling **Disable Receive Shadows** on the material correctly disables shadow receiving for that material for all light types.

**Comment by jsjtxietian** [comment]
Thanks! I will try not to update my llvm too aggressively next time :(

**Comment by akien-mga** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.2.1.

**Pull request merged**